gui/launcher.h

```diff
@@ -124,11 +155,77 @@ class LauncherDialog : public Dialog {
 	 *
 	 * @target	name of target to select
 	 */
-	void selectTarget(const String &target);
+	virtual void selectTarget(const String &target) = 0;
+	virtual int getSelected() = 0;
 private:
 	bool checkModifier(int modifier);
 };
 
+// Disable the grid for platforms that disable fancy themes
+class LauncherChooser {
+protected:
+	LauncherDialog *_impl;
+
+public:
+	LauncherChooser();
+	~LauncherChooser();
+
+	int runModal();
+	void selectLauncher();
+};
+
+class LauncherSimple : public LauncherDialog {
+public:
+	LauncherSimple(const U32String &title);
+
+	void handleCommand(CommandSender *sender, uint32 cmd, uint32 data) override;
+	void handleKeyDown(Common::KeyState state) override;
+
+#ifndef DISABLE_LAUNCHERDISPLAY_GRID
```

Comment:
I think we can retain this bit even when disabling these.

Reply:
I have removed unnecessary directives in 215ddc4.